Add recipe for votd

kristjoc opened the pull request:
### Brief summary of what the package does

votd is a simple Emacs package that fetches the Bible verse of the
day from https://www.biblegateway.com/ and formats it to be used as
an emacs-dashboard footer or *scratch* buffer message in Emacs.

### Direct link to the package repository

https://github.com/kristjoc/votd

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  I think the idea for this package is fine, bt I do note there is another similar package in this space, i.e. https://melpa.org/#/insert-esv - I don't think it includes a "daily" verse, but maybe there is some change to share ideas there.

In the meantime, quick first pass -

⸺ `votd.el` with byte-compile using Emacs 30.1:
```
votd.el:136:10: Warning: Unused lexical variable `url-queue-timeout'
```

⸺ `votd.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- votd.el#L159: No `format` required; `error` takes an f-string
- votd.el#L171: No `format` required; `error` takes an f-string
```

⸺ `votd.el` with checkdoc 0.6.2 (fix *within reason*):
```
votd.el:154: White space found at end of line
```

⸺ Package and license:
- Please specify `:fetcher` before `:repo` in your recipe

kristjoc commented:
Many thanks for the feedback!

> Thanks for this. I think the idea for this package is fine, bt I do note there is another similar package in this space, i.e. https://melpa.org/#/insert-esv - I don't think it includes a "daily" verse, but maybe there is some change to share ideas there.
> 

I was aware of that package, but it differs from votd because of the following:

- insert-esv is bound to ESV translation only
- it requires users to create a Crossway account and obtain an API key
- users must specify a verse in order to retrieve the text

> In the meantime, quick first pass -
> 
> ⸺ `votd.el` with byte-compile using Emacs 30.1:
> 
> ```
> votd.el:136:10: Warning: Unused lexical variable `url-queue-timeout'
> ```
> 

Fixed

> ⸺ `votd.el` with [melpazoid](https://github.com/riscy/melpazoid):
> 
> ```
> - votd.el#L159: No `format` required; `error` takes an f-string
> - votd.el#L171: No `format` required; `error` takes an f-string
> ```
> 

Fixed

> ⸺ `votd.el` with checkdoc 0.6.2 (fix _within reason_):
> 
> ```
> votd.el:154: White space found at end of line
> ```
> 

Fixed

> ⸺ Package and license:
> 
> * Please specify `:fetcher` before `:repo` in your recipe

Fixed

riscy commented:
Thanks - and welcome to MELPA. :)
